sdk/python/feast/repo_config.py

```diff
@@ -82,45 +105,84 @@ def _validate_online_store_config(cls, values):
         # granular configuration system
 
-        # Skip if online store isn't set explicitly
+        # Set empty online_store config if it isn't set explicitly
         if "online_store" not in values:
             values["online_store"] = dict()
 
-        # Skip if we arent creating the configuration from a dict
+        # Skip if we aren't creating the configuration from a dict
         if not isinstance(values["online_store"], Dict):
             return values
 
         # Make sure that the provider configuration is set. We need it to set the defaults
         assert "provider" in values
 
-        if "online_store" in values:
-            # Set the default type
-            if "type" not in values["online_store"]:
-                if values["provider"] == "local":
-                    values["online_store"]["type"] = "sqlite"
-                elif values["provider"] == "gcp":
-                    values["online_store"]["type"] = "datastore"
-
-            online_store_type = values["online_store"]["type"]
-
-            # Make sure the user hasn't provided the wrong type
-            assert online_store_type in ["datastore", "sqlite"]
-
-            # Validate the dict to ensure one of the union types match
-            try:
-                if online_store_type == "sqlite":
-                    SqliteOnlineStoreConfig(**values["online_store"])
-                elif values["online_store"]["type"] == "datastore":
-                    DatastoreOnlineStoreConfig(**values["online_store"])
-                else:
-                    raise ValidationError(
-                        f"Invalid online store type {online_store_type}"
-                    )
-            except ValidationError as e:
+        # Set the default type
+        if "type" not in values["online_store"]:
+            if values["provider"] == "local":
+                values["online_store"]["type"] = "sqlite"
+            elif values["provider"] == "gcp":
+                values["online_store"]["type"] = "datastore"
+
+        online_store_type = values["online_store"]["type"]
+
+        # Make sure the user hasn't provided the wrong type
+        assert online_store_type in ["datastore", "sqlite"]
+
+        # Validate the dict to ensure one of the union types match
+        try:
+            if online_store_type == "sqlite":
+                SqliteOnlineStoreConfig(**values["online_store"])
+            elif online_store_type == "datastore":
+                DatastoreOnlineStoreConfig(**values["online_store"])
+            else:
+                raise ValidationError(f"Invalid online store type {online_store_type}")
+        except ValidationError as e:
+            raise ValidationError(
+                [ErrorWrapper(e, loc="online_store")], model=SqliteOnlineStoreConfig,
+            )
+
+        return values
+
+    @root_validator(pre=True)
+    def _validate_offline_store_config(cls, values):
```

Comment:
Damn I have created a mess with Pydantic. We'll need to figure out a clean way to fix this.